### PR TITLE
Added Cypress for E2E testings

### LINKED ISSUE #77: Setup Cypress for E2E testings
### Is your enhancement related to a problem? Please describe.

As we're switching to use Cypress for E2E testings, and ultimately replace WP Acceptance, it's great to start implementing Cypress E2E testings for this plugin.

Some of 10up open-source plugins using Cypress:
- https://github.com/10up/restricted-site-access
- https://github.com/10up/simple-podcasting

Other useful links:
- 10up Open Source Best Practices for Testing: https://10up.github.io/Open-Source-Best-Practices/testing/
- https://github.com/10up/cypress-wp-setup/ && https://github.com/10up/cypress-wp-utils will be helpful in getting this spun up and some initial test commands to leverage.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by iamdharmesh**
### Description of the Change
This PR removes wpacceptance tests and Adds end-to-end testing with Cypress to follow [OSBP guide](https://10up.github.io/Open-Source-Best-Practices/testing/#e2e-testing). Test runs automatically on every pull request to the `develop` branch.

Steps to perform local testing or develop new tests:

- Install dependencies `npm install`
- Start wp-env local development suite `npm run env:start`
- Run cypress `npm run cypress:run` or open developer browser `npm run cypress:open`

**Added Tests:**

- Activate the plugin.
- Plugin is enabled for page.
- Change parent pages order.
- Change child pages order.
- Change posts order.

Closes #77

### Alternate Designs
n/a

### Possible Drawbacks
n/a

### Verification Process
GitHub Actions workflow "E2E test", expected to pass the test. Refer to "Description of the Change" for local testing.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Credits

Props @iamdharmesh

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: E2E test
+
+on:
+  pull_request:
+    branches:
+      - develop
```
Comment (faisal-alvi) [inline .github/workflows/cypress.yml]: Shouldn't we test it on push to: trunk and develop too?

File: tests/cypress/integration/admin.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+describe("Admin can login and make sure plugin is activated", () => {
```
Comment (faisal-alvi) [inline tests/cypress/integration/admin.test.js]:
```suggestion
describe('Admin can login and make sure plugin is activated', () => {
```

File: tests/cypress/integration/admin.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	it("Can activate plugin if it is deactivated", () => {
```
Comment (faisal-alvi) [inline tests/cypress/integration/admin.test.js]:
```suggestion
	it('Can activate plugin if it is deactivated', () => {
```

File: tests/cypress/integration/admin.test.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		cy.login();
+		cy.visit("/wp-admin/plugins.php");
```
Comment (faisal-alvi) [inline tests/cypress/integration/admin.test.js]:
```suggestion
		cy.visit('/wp-admin/plugins.php');
```

File: tests/cypress/integration/admin.test.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+				cy.get('#the-list tr#simple-page-ordering .deactivate > a').click();
+				cy.get('#the-list tr#simple-page-ordering .activate > a').click();
+				cy.get('#the-list tr#simple-page-ordering .deactivate > a').should('have.text', 'Deactivate');
+			}
+		});
+	});
+
+	it('Can see "Sort by Order" on Pages list page.', () => {
+		cy.login();
+		cy.visit("/wp-admin/edit.php?post_type=page");
```
Comment (faisal-alvi) [inline tests/cypress/integration/admin.test.js]:
```suggestion
		cy.visit('/wp-admin/edit.php?post_type=page');
```

File: tests/cypress/integration/page-ordering.test.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+describe("Test Page Order Change", () => {
```
Comment (faisal-alvi) [inline tests/cypress/integration/page-ordering.test.js]: Please use single quotes wherever applicable in all test files.

File: tests/cypress/integration/post-ordering.test.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+describe("Test Post Order Change", () => {
```
Comment (faisal-alvi) [inline tests/cypress/integration/post-ordering.test.js]: Please use single quotes wherever applicable in all test files.

File: tests/cypress/integration/page-ordering.test.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+	it("Can change parent pages order", () => {
+		cy.login();
+		cy.visit("/wp-admin/edit.php?post_type=page");
+
+		const first = '.wp-list-table tbody tr:nth-child(1)';
+		const second = '.wp-list-table tbody tr:nth-child(2)';
+		let firstText = cy.get(`${first} .row-title`);
+		let secondText = cy.get(`${second} .row-title`);
```
Comment (faisal-alvi) [inline tests/cypress/integration/page-ordering.test.js]: Please use `const` as already used below.

File: tests/cypress/integration/admin.test.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+describe("Admin can login and make sure plugin is activated", () => {
+	it("Can activate plugin if it is deactivated", () => {
+		cy.login();
+		cy.visit("/wp-admin/plugins.php");
+
+		cy.get( 'body' ).then( ( $body ) => {
+			if( $body.find( '#the-list tr[data-slug="simple-page-ordering"] .deactivate > a' ).length > 0 ) {
```
Comment (faisal-alvi) [inline tests/cypress/integration/admin.test.js]: Are we following 'no space' before and after the `body/$body` and inside the `if` condition? OR are we following one space on both sides? Looking at the rest of the code, looks like we do not leave spaces.

File: tests/cypress/integration/page-ordering.test.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+		
+		cy.get(`${first} .row-title`).then($el => {
+			secondText.should('have.text', $el.text());
+		});
+		cy.get(`${second} .row-title`).then($el => {
+			firstText.should('have.text', $el.text());
+		});
+	  });
+});
+
```
Comment (faisal-alvi) [inline tests/cypress/integration/page-ordering.test.js]: The extra blank line can be removed at the EOF.

File: tests/cypress/integration/post-ordering.test.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+	  	cy.get(`${first} .row-title`).then($el => {
+			secondText.should('have.text', $el.text());
+		});
+		cy.get(`${second} .row-title`).then($el => {
+			firstText.should('have.text', $el.text());
+		});
+	});
+});
+
```
Comment (faisal-alvi) [inline tests/cypress/integration/post-ordering.test.js]: The extra blank line can be removed at the EOF.

**Review by faisal-alvi — COMMENTED** [review]
@iamdharmesh really great efforts! Just added a few minor coding standard related comments. Please have a look.

File: .github/workflows/cypress.yml (hunk 1 above)
Comment (iamdharmesh) [reply .github/workflows/cypress.yml]: Yes, we should. Good catch. done.
I will raise an issue in wp-cypress-setup lib

File: tests/cypress/integration/page-ordering.test.js (hunk 8 above)
Comment (iamdharmesh) [reply tests/cypress/integration/page-ordering.test.js]: Nice catch.

File: tests/cypress/integration/page-ordering.test.js (hunk 10 above)
Comment (iamdharmesh) [reply tests/cypress/integration/page-ordering.test.js]: As per rule here https://github.com/10up/simple-page-ordering/blob/develop/.editorconfig#L9, new line will be always there at the EOF

File: .github/workflows/cypress.yml (hunk 1 above)
Comment (iamdharmesh) [reply .github/workflows/cypress.yml]: @cadic do you think should we add this in cypress-wp-setup lib?

**Review by faisal-alvi — APPROVED** [review]
🚀 🦸🏻

**Pull request merged**